crates/utils/src/settings/structs.rs

```diff
@@ -237,3 +233,14 @@ pub struct PrometheusConfig {
   #[doku(example = "10002")]
   pub port: i32,
 }
+
+#[derive(Debug, Deserialize, Serialize, Clone, SmartDefault, Document)]
+#[serde(default)]
+// named federation"worker"config to disambiguate from the activitypub library configuration
+pub struct FederationWorkerConfig {
```

Comment:
Its a bit useless to make a new struct for a single value.

Reply:
To be honest some settings like federation enabled or signed fetch would also fit better in the config, as they require a restart anyway. Wasnt a good idea to move everything into the db back then.

Reply:
When updating federation config in db, it might be possible to use the postgresql notify command to tell lemmy server processes to restart the federation workers, and use the listen command and [poll_message](https://docs.rs/tokio-postgres/0.7.10/tokio_postgres/struct.Connection.html#method.poll_message) to respond to the notifications

Reply:
Ah, i wasn't even aware of those DB settings that are related to this one. In any case I think I'll leave it for now, the struct isn't expensive or hard to read or anything and has a nice attached doc-string.